packages/plugin-ext-vscode/src/node/plugin-vscode-file-handler.ts

```diff
@@ -50,34 +39,45 @@ export class PluginVsCodeFileHandler implements PluginDeployerFileHandler {
-        const extensionDir = await this.getExtensionDir(context);
-        console.log(`[${id}]: trying to decompress into "${extensionDir}"...`);
-        if (context.pluginEntry().type === PluginType.User && await fs.pathExists(extensionDir)) {
-            console.log(`[${id}]: already found`);
-            context.pluginEntry().updatePath(extensionDir);
+        const id = this.getNormalizedExtensionId(context.pluginEntry().id());
+        const extensionDeploymentDir = await this.getExtensionDeploymentDir(id);
+        try {
+            console.log(`[${id}]: trying to decompress into "${extensionDeploymentDir}"...`);
+
+            if (await fs.pathExists(extensionDeploymentDir)) {
+                console.log(`[${id}]: deployment dir "${extensionDeploymentDir}" already exists ...`);
+                context.pluginEntry().updatePath(extensionDeploymentDir);
+                return;
+            }
+
+            if (!await decompressExtension(context.pluginEntry().path(), extensionDeploymentDir)) {
+                console.error(`[${id}]: decompression failed`);
+                return;
+            }
+
+            console.log(`[${id}]: decompressed`);
+        } catch (error) {
+            console.error(`[${id}]: error while decompressing`, error);
             return;
         }
-        await this.decompress(extensionDir, context);
-        console.log(`[${id}]: decompressed`);
-        context.pluginEntry().updatePath(extensionDir);
-    }
 
-    protected async getExtensionDir(context: PluginDeployerFileHandlerContext): Promise<string> {
-        const systemExtensionsDirUri = await this.systemExtensionsDirUri.promise;
-        return FileUri.fsPath(systemExtensionsDirUri.resolve(filenamify(context.pluginEntry().id(), { replacement: '_' })));
+        console.log(`[${id}]: decompressed to temp dir "${extensionDeploymentDir}"`);
+        context.pluginEntry().updatePath(extensionDeploymentDir);
     }
 
-    protected async decompress(extensionDir: string, context: PluginDeployerFileHandlerContext): Promise<void> {
-        await context.unzip(context.pluginEntry().path(), extensionDir);
-        if (context.pluginEntry().path().endsWith('.tgz')) {
-            const extensionPath = path.join(extensionDir, 'package');
-            const vscodeNodeModulesPath = path.join(extensionPath, 'vscode_node_modules.zip');
-            if (await fs.pathExists(vscodeNodeModulesPath)) {
-                await context.unzip(vscodeNodeModulesPath, path.join(extensionPath, 'node_modules'));
-            }
+    protected async getExtensionDeploymentDir(pluginId: string): Promise<string> {
+        try {
+            const deployedPluginsDirUri = await this.environment.getDeploymentDirUri();
+            const extensionDeploymentDirPath = FileUri.fsPath(deployedPluginsDirUri.resolve(pluginId));
+            return extensionDeploymentDirPath;
+        } catch (error) {
+            console.error('Error resolving extension directory:', error);
```

Comment:
If the error is unhandled, we'll get a stack trace. If it isn't shouldn't the handler print the error?